### ISSUE TITLE
hashtags wikirule breaks stylesheets

### ISSUE THREAD

**Issue by tobibeer** [TiddlyWiki/TiddlyWiki5#2192]
I was about to write a wikirule parser that will do something with a **#hashtag**, or so I thought.

Turns out, this appears to break about every single stylesheet tiddler or palette being loaded by TiddlyWiki, except those of type `text/css`, I presume.

How can I possibly cater for a **#hashtag** wikirule and make sure it never gets applied for tiddlers matching a given filter, e.g. never those tagged `$:/tags/Stylesheet`, and never to palettes?

Furthermore, I don't see any capability to do:

```
\rules allow hashtag
```

...to allow the rule for a given context only, e.g. things within the PageTemplate. Actually, I am not quite sure where stylesheets get loaded at all.

Let's say a Stylesheet retrieves a color from a palette:

```
{{SomePalette##some-color}}
```

...or uses the color macro to achieve the same.

How can I ensure it will never use the hashtag wikirule to parse the contents so that the result is never going to be hash-tagified?

Atm I feel like it just isn't possible to separate content (hashtags) from presentation (styles) that way.

**Comment by pmario** [comment]
There are some discussions, to improve the tw tiddler "type" mechanism with the possibility to enable additional or completely different wikitext vocabularies. see issue  #345:  "Introduce vocabulary parameter to wikitext content type"

@buggyj also created a working prototype: http://tw5vocab.tiddlyspot.com/ ... So having a closer look at this implementation imo would also solve some other long standing issues.

**Comment by tobibeer** [comment]
Hi @pmario, as for #345 I definitely do **not** want (users) to have to be specific about the vocab. In terms of vocab terminology, I guess I'd want a different global vocabulary for different aspects contexts of the core, e.g. "everything stylesheet / theme" is parsed with a different vocab than "content"... however those two get to be separated.

**Comment by tobibeer** [comment]
It feels like **all core stylesheets**, palettes, themes need to explicitly define:

```
\rules only this and that
```

...so as to be more flexible in what is allowed in actual content without breaking a theme or stylesheet.

**Comment by tobibeer** [comment]
@Jermolene, I am starting to think that whenever the text of a tiddler tagged **$:/tags/Stylesheet** or an index of a tiddler tagged **$:/tags/Palette** is parsed and transcluded, the result must never contain html ...since, clearly, we are operating on what is expected to be valid css syntax in terms of output.

I don't really know, but it feels like this requires wiki-rules to operate in either a dom-mode or a text-mode, whereas the latter must never yield widgets that create html elements in the dom.

In other words, the parser identifying a wikirule handling a something like a `#hashtag` — a pattern which color entries in the palette may as well match, e.g. `#ABC`, `#BBC`, `#CAB` — may normally render a widget. However, when in text mode, it will only ever retrieve a text-node at best.

**Comment by twMat** [comment]
Likely not of practical use but anyway FYI: There are a few variants of the hashtag, or *number sign* as the characters "real" name apparently is, 井 ♯ ⌗ ⋕ # - see [wikipedia](https://en.wikipedia.org/wiki/Number_sign)

**Comment by pmario** [comment]
Same as: https://github.com/TiddlyWiki/TiddlyWiki5/issues/2191
